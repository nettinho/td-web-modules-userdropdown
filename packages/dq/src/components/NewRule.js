import _ from "lodash/fp";
import React from "react";
import PropTypes from "prop-types";
import { bindActionCreators } from "redux";
import { connect } from "react-redux";
import { Container, Header, Icon, Segment } from "semantic-ui-react";
import { FormattedMessage } from "react-intl";
import { createRule } from "../routines";
import RuleForm from "./RuleForm";

export const NewRule = ({ rule, createRule }) => (
  <Container as={Segment} text>
    <Header as="h2">
      <Icon name="wrench" />
      <Header.Content>
        {!_.isEmpty(rule) && <FormattedMessage id="rule.actions.create" />}
        {_.isEmpty(rule) && <FormattedMessage id="rules.actions.create" />}
      </Header.Content>
    </Header>
    <RuleForm handleSubmit={createRule} />
  </Container>
);

NewRule.propTypes = {
  rule: PropTypes.object,
  createRule: PropTypes.func.isRequired
};

const mapStateToProps = ({ rule }) => ({ rule });
const mapDispatchToProps = dispatch => ({
  ...bindActionCreators({ createRule }, dispatch)
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(NewRule);
