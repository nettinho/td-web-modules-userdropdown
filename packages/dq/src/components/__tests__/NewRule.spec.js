import React from "react";
import { shallow } from "enzyme";
import { NewRule } from "../NewRule";

describe("<NewRule />", () => {
  const createRule = jest.fn();
  const rule = { id: 1, name: "nn", description: "dd" };
  const props = { rule, createRule };

  it("matches the latest snapshot", () => {
    const wrapper = shallow(<NewRule {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
