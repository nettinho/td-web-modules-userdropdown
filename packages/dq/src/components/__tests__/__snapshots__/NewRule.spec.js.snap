// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NewRule /> matches the latest snapshot 1`] = `
<Container
  as={[Function]}
  text={true}
>
  <Header
    as="h2"
  >
    <Icon
      as="i"
      name="wrench"
    />
    <HeaderContent>
      <FormattedMessage
        id="rule.actions.create"
        values={Object {}}
      />
    </HeaderContent>
  </Header>
  <withRouter(InjectIntl(Connect(RuleForm)))
    handleSubmit={[MockFunction]}
  />
</Container>
`;
